projects/lights_control/src/lights_config.c

```diff
@@ -0,0 +1,103 @@
+#include "misc.h"
+
+#include "gpio.h"
+#include "lights_config.h"
+#include "lights_events.h"
+
+#define CONVERT_TO_MASK(index) 1 << index
+
+static const GPIOAddress s_board_type_address = { .pin = 13, .port = GPIO_PORT_B };
+
+static const GPIOAddress s_addresses_front[] = {
+  [LIGHTS_CONFIG_FRONT_LIGHT_HORN] = { .port = GPIO_PORT_B, .pin = 11 },                  //
+  [LIGHTS_CONFIG_FRONT_LIGHT_HIGH_BEAMS_RIGHT] = { .port = GPIO_PORT_B, .pin = 1 },       //
+  [LIGHTS_CONFIG_FRONT_LIGHT_HIGH_BEAMS_LEFT] = { .port = GPIO_PORT_B, .pin = 15 },       //
+  [LIGHTS_CONFIG_FRONT_LIGHT_LOW_BEAMS_RIGHT] = { .port = GPIO_PORT_B, .pin = 2 },        //
+  [LIGHTS_CONFIG_FRONT_LIGHT_LOW_BEAMS_LEFT] = { .port = GPIO_PORT_A, .pin = 8 },         //
+  [LIGHTS_CONFIG_FRONT_LIGHT_DRL_RIGHT] = { .port = GPIO_PORT_B, .pin = 0 },              //
+  [LIGHTS_CONFIG_FRONT_LIGHT_DRL_LEFT] = { .port = GPIO_PORT_A, .pin = 10 },              //
+  [LIGHTS_CONFIG_FRONT_LIGHT_SIDE_LEFT_INDICATOR] = { .port = GPIO_PORT_B, .pin = 14 },   //
+  [LIGHTS_CONFIG_FRONT_LIGHT_LEFT_TURN] = { .port = GPIO_PORT_A, .pin = 9 },              //
+  [LIGHTS_CONFIG_FRONT_LIGHT_SIDE_RIGHT_INDICATOR] = { .port = GPIO_PORT_B, .pin = 12 },  //
+  [LIGHTS_CONFIG_FRONT_LIGHT_RIGHT_TURN] = { .port = GPIO_PORT_A, .pin = 10 },            //
+};
+
+static const GPIOAddress s_addresses_rear[] = {
+  [LIGHTS_CONFIG_REAR_LIGHT_STROBE] = { .port = GPIO_PORT_B, .pin = 11 },            //
+  [LIGHTS_CONFIG_REAR_LIGHT_RIGHT_BRAKE] = { .port = GPIO_PORT_B, .pin = 1 },        //
+  [LIGHTS_CONFIG_REAR_LIGHT_RIGHT_OUTER_BRAKE] = { .port = GPIO_PORT_B, .pin = 2 },  //
+  [LIGHTS_CONFIG_REAR_LIGHT_LEFT_BRAKE] = { .port = GPIO_PORT_B, .pin = 15 },        //
+  [LIGHTS_CONFIG_REAR_LIGHT_LEFT_OUTER_BRAKE] = { .port = GPIO_PORT_A, .pin = 8 },   //
+  [LIGHTS_CONFIG_REAR_LIGHT_CENTRE_BRAKE] = { .port = GPIO_PORT_B, .pin = 14 },      //
+  [LIGHTS_CONFIG_REAR_LIGHT_LEFT_OUTER_TURN] = { .port = GPIO_PORT_A, .pin = 10 },   //
+  [LIGHTS_CONFIG_REAR_LIGHT_LEFT_TURN] = { .port = GPIO_PORT_A, .pin = 9 },          //
+  [LIGHTS_CONFIG_REAR_LIGHT_RIGHT_OUTER_TURN] = { .port = GPIO_PORT_B, .pin = 0 },   //
+  [LIGHTS_CONFIG_REAR_LIGHT_RIGHT_TURN] = { .port = GPIO_PORT_B, .pin = 10 },        //
+};
+
+static const GPIOSettings s_gpio_settings_out = { .direction = GPIO_DIR_OUT,
```

Comment:
Do these really need to be here?

Reply:
I guess not, they're not containers of any sort. No point having them as static variables.